### PR TITLE
t2 run: move JS-specific "post run" operation to a language specific definition

### PR DESCRIPTION
**Pull request by rwaldron**
@johnnyman727 This should be reviewed and merged before any of the pending Rust support changes.

Signed-off-by: Rick Waldron <waldron.rick@gmail.com>

### CODE REVIEW

File: lib/tessel/deployment/rust.js [hunk 1]
```diff
@@ -60,5 +60,6 @@ exportables.preRun = function(tessel) {
   });
 };
 
+// exportables.postRun = function(tessel, remote) {};
```
Comment (rwaldron) [inline lib/tessel/deployment/rust.js]: This is here as "pseudo documentation" until the language support API churn slows and we can start documenting properly

File: lib/tessel/deployment/javascript.js [hunk 2]
```diff
@@ -68,6 +68,17 @@ exportables.preBundle = function(opts) {
   return exportables.resolveBinaryModules(opts);
 };
 
+// exportables.preRun = function(tessel) {};
+
+exportables.postRun = function(tessel, remote) {
```
Comment (johnnyman727) [inline lib/tessel/deployment/javascript.js]: It's not great that the arguments for `preRun`, `preBundle, `tarBundle`, and `postRun` all have very different arguments. Is there any way we can make them more consistent and intuitive?

Comment (rwaldron) [inline lib/tessel/deployment/javascript.js]: I'm not sure that holds up very well, considering:

- There is no such thing as a "remote process" when `preRun` is called, but would also be irrelevant to that operation.
- `preBundle` and `tarBundle` are about files and directories, not Tessel or "remote process". 


The best that I think can be done is:

- `preBundle(options)` 
- `tarBundle(options)` 
- `preRun(tessel, options)`
- `postRun(tessel, options)` (`options` will contain `remoteProcess`)

File: lib/tessel/deploy.js [hunk 3]
```diff
@@ -355,23 +355,24 @@ exportables.sendBundle = function(tessel, opts) {
...
-    if (opts.lang.preRun) {
-      prom = opts.lang.preRun(tessel, opts);
+    if (lang.preRun) {
+      preRun = lang.preRun(tessel, options);
     }
 
-    return prom.then(() => {
+    return preRun.then(() => {
       tessel.connection.exec(
-        commands[opts.lang.meta.extname].execute(Tessel.REMOTE_RUN_PATH, opts.resolvedEntryPoint), {
```
Comment (rwaldron) [inline lib/tessel/deploy.js]: This is just renaming, nothing interesting

File: lib/tessel/deploy.js [hunk 4]
```diff
@@ -355,23 +355,24 @@ exportables.sendBundle = function(tessel, opts) {
...
+        commands[lang.meta.extname].execute(Tessel.REMOTE_RUN_PATH, options.resolvedEntryPoint), {
```
Comment (rwaldron) [inline lib/tessel/deploy.js]: This is just renaming, nothing interesting

File: lib/tessel/deployment/javascript.js [hunk 5]
```diff
@@ -63,11 +63,21 @@ var exportables = {
   lists: lists,
 };
 
-
-exportables.preBundle = function(opts) {
-  return exportables.resolveBinaryModules(opts);
+/*
+exportables.preRun = function(tessel, options) {
+  return Promise.resolve();
 };
```
Comment (rwaldron) [inline lib/tessel/deployment/javascript.js]: This is not removed, just relocated next to `exportables.tarBundle`

**Comment by johnnyman727** [comment]
LGTM, I think the API changes are as good as can be.

**Pull request merged**